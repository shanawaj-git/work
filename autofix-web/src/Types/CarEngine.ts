export interface CarEngineQueryResponseType {
  id: string;
  attributes: {
    name: string;
  };
}

export interface CarEngineType {
  id: string;
  name: string;
}
