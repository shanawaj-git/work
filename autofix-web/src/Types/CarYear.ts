export interface CarYearQueryResponseType {
  id: string;
  attributes: {
    value: number;
  };
}

export interface CarYearsQueryResponseWrapperType {
  years: {
    data: CarYearQueryResponseType[];
  };
}

export interface CarYearType {
  id: string;
  value: number;
}
