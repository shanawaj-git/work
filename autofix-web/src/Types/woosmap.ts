export interface windowMultiSearch {
  woosmap: {
    multisearch: (e: woosmapMutliSearchPayloadType) => multiSearchType;
  };
}

export interface processAutoCompleteType {
  addressComponents?: string[];
  label?: string;
  id: string;
  api: string;
}

export interface getPlaceDetailsType {
  id: string;
  api: string;
}

export interface multiSearchType {
  autocompleteMulti: (
    e: string
  ) => Promise<autocompleteMultiResponse[] | string>;
  detailsMulti: (obj: getPlaceDetailsType) => Promise<detailsMultiType>;
}

export interface autocompleteMultiResponse {
  id: string;
  api: string;
  item: autocompleteMultiResponseItem;
  description: string;
  highlight: string;
  score: number | string;
  types: string[];
}

interface autocompleteMultiResponseItem {
  terms: autocompleteMultiResponseItemTerms[];
  description: string;
}

interface autocompleteMultiResponseItemTerms {
  value: string;
}

export interface woosmapMutliSearchPayloadType {
  apiOrder?: string[];
  localities?: Localities;
  address?: addressType;
  places?: placesType;
  components?: componentType;
}

export interface paramsTypes {
  language: string;
  components: componentType;
}

export interface componentType {
  country: string[];
}
export interface Localities {
  key: string;
  params: paramsTypes;
}

export interface addressType {
  key: string;
  params: paramsTypes;
}

export interface placesType {
  params: paramsTypes;
  key: string;
}

export interface Location {
  lat: number;
  lng: number;
}

export interface geometryType {
  location: Location;
}

export interface addressComponentType {
  long_name: string;
  short_name: string;
  types: string[];
}

export interface itemType {
  address_components: addressComponentType[];
  adr_address: string;
  formatted_address: string;
  geometry: geometryType;
  icon: string;
  name: string;
  place_id: string;
  types: string[];
  url: string;
  vicinity: string;
  html_attributions?: unknown;
}

export interface detailsMultiType {
  formatted_address?: string;
  geometry: geometryType;
  name?: string;
  api?: string;
  types?: string[];
  id?: string;
  address_components?: addressComponentType[];
  item?: itemType;
}
