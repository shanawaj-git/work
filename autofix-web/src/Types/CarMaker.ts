import { CarModelType, CarModelTypeQueryResponse } from "./CarModel";

export interface CarMakerType {
  id: string;
  name: string;
  logo: string;
  models?: Array<CarModelType>;
}

export interface CarMakerTypeQueryResponse {
  id: string;
  attributes: {
    logo: string;
    name: string;
    models?: {
      data: Array<CarModelTypeQueryResponse>;
    };
  };
}
