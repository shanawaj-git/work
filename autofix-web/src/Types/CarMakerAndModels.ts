import { CarMakerTypeQueryResponse } from "./CarMaker";

export interface CarMakerAndModelsQueryResponse {
  makers: {
    data: Array<CarMakerTypeQueryResponse>;
  };
}

export interface FlattenedMakerAndModel {
  id: string;
  name: string;
  image?: string;
  makerModelName: string;
  makerName: string;
}
