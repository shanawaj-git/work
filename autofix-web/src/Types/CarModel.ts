import { CarMakerTypeQueryResponse } from "./CarMaker";
export interface CarModelType {
  id: string;
  name: string;
  image?: string;
}

export interface CarModelTypeQueryResponse {
  id: string;
  attributes: {
    image?: string;
    name: string;
    makerId?: {
      data: CarMakerTypeQueryResponse;
    };
  };
}
