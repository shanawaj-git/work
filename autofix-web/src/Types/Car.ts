import { CarMakerType } from "./CarMaker";
import { CarYearType, CarYearQueryResponseType } from "./CarYear";
import { CarModelType, CarModelTypeQueryResponse } from "./CarModel";
import { CarEngineQueryResponseType, CarEngineType } from "./CarEngine";

export interface CarsQueryResponseWrapperType {
  cars: {
    data: CarQueryResponseType[];
  };
}

export interface CarQueryResponseType {
  attributes: {
    engine_type: {
      data: CarEngineQueryResponseType;
    };
    model: {
      data: CarModelTypeQueryResponse;
    };
    year: {
      data: CarYearQueryResponseType;
    };
    image?: string;
  };
}

export interface CarType {
  maker: CarMakerType;
  model: CarModelType;
  year: CarYearType;
  engineType: CarEngineType;
  image: string;
}
