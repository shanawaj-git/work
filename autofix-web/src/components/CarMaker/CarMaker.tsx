import React, { FC } from "react";
import { utils } from "@albathanext/design-system";
import { CarModelType } from "../../Types/CarModel";
import { CarMakerType } from "../../Types/CarMaker";
import ItemCardWithAction from "../ItemCardWithAction/ItemCardWithAction";

interface CarMakerProps {
  data?: CarMakerType;
  testId?: string;
  className?: string;
  onModelSelection: (makerId: string, modelId: string) => void;
}

const CarMaker: FC<CarMakerProps> = ({ ...props }: CarMakerProps) => {
  const { testId, className, data, onModelSelection } = props;
  if (!data) return <></>;
  return (
    <div
      className={"pt-6".concat(" ", className || "")}
      data-testid={`CarMaker-${testId || data.id}`}
    >
      <div className="flex items-center justify-between">
        <p
          className="not-italic	font-[500] leading-5.5 text-[17px]"
          style={{
            borderColor: utils.colorSelector("P300"),
          }}
        >
          {data.name}
        </p>
        <div>
          <img
            alt={data.name}
            className="h-5.25 w-9 left-1.25 top-4.5 rounded-none"
            src={`${data.logo}`}
          />
        </div>
      </div>
      {props.data &&
        props.data.models &&
        props.data.models.map((carModel: CarModelType) => (
          <ItemCardWithAction
            key={carModel.id}
            itemText={carModel.name}
            itemId={carModel.id}
            onClick={() => onModelSelection(data.id, carModel.id)}
          />
        ))}
    </div>
  );
};

export default CarMaker;
