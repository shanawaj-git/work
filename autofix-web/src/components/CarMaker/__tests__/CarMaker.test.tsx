import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CarMaker from "../CarMaker";
import { CarMakerType } from "../../../Types/CarMaker";

describe("<CarMaker />", () => {
  const carMakerData: CarMakerType = {
    id: "1",
    name: "Toyota",
    logo: "url",
    models: [
      {
        id: "1",
        name: "Corolla",
        image: "",
      },
    ],
  };

  test("it should not return car maker when given data in empty", () => {
    const { container } = render(<CarMaker data={null} />);

    expect(container).toBeEmptyDOMElement();
  });

  test("it should mount", () => {
    render(<CarMaker data={carMakerData} />);

    const component = screen.getByTestId(`CarMaker-${carMakerData.id}`);

    expect(component).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { container } = render(<CarMaker data={carMakerData} />);

    expect(container).toMatchSnapshot();
  });
});
