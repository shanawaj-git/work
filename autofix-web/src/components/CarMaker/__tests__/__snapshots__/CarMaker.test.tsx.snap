// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarMaker /> it should match snapshot 1`] = `
<div>
  <div
    class="pt-6 "
    data-testid="CarMaker-1"
  >
    <div
      class="flex items-center justify-between"
    >
      <p
        class="not-italic font-[500] leading-5.5 text-[17px]"
        style="border-color: #e25f6d;"
      >
        Toyota
      </p>
      <div>
        <img
          alt="Toyota"
          class="h-5.25 w-9 left-1.25 top-4.5 rounded-none"
          src="url"
        />
      </div>
    </div>
    <div
      class="flex p-5 mt-3 items-center justify-between rounded border cursor-pointer border-solid border-[#00000014]"
      data-testid="itemCardWithAction-1"
      style="gap: 10px; background-color: rgb(255, 255, 255);"
    >
      <p
        class="font-normal font-[500] text-base leading-6"
      >
        Corolla
      </p>
      <div>
        <img
          alt="forward action icon"
          class="min-h-3.5 min-w-3 left-1.25 top-4.5 rounded-none"
          src="arrow-forward-icon.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
